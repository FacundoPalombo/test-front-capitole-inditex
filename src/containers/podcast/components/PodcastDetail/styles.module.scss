.podcast_detail {
  width: auto;
  height: auto;
  display: inline-block;
  box-sizing: content-box;

  &_navigable {
    text-decoration: none;
    color: inherit;
    &:visited {
      color: inherit;
    }
  }

  &__card {
    box-shadow: 0px 1px 4px 1px var(--shadow);
    display: flex;
    align-items: flex-start;
    justify-content: flex-start;
    flex-direction: column;
    width: 250px;
    padding: 8px;
    overflow: visible;
    margin-top: auto;
    border-radius: 4px;
    box-sizing: border-box;
  }
  &__picture {
    border-radius: 4px;
    width: 170px;
    height: 170px;
    align-self: center;
  }
  &__separator {
    color: #aaa;
    width: 95%;
    opacity: 0.35;
    border: 1px solid;
  }
  &__title {
    color: var(--font-primary);
    font-weight: 600;
    margin-left: 16px;
    margin-block-start: 0.5em;
    margin-block-end: 0.5em;
  }
  &__artist {
    margin-left: 16px;
  }
  &__description {
    text-align: start;
    padding: 0 8px;
    width: 100%;
    box-sizing: border-box;
    word-wrap: break-word;
    text-overflow: clip;
    white-space: break-spaces;
    &_empathized {
      width: 100%;
      display: block;
      box-sizing: border-box;
      padding: 5px;
    }
    &_strong {
      width: 100%;
      display: block;
      margin-bottom: 16px;
    }
  }
}
