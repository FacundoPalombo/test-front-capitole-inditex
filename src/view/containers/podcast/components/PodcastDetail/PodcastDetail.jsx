import React from 'react';
import styles from './styles.module.scss';
import { string } from 'prop-types';

const PodcastDetail = ({ image, title, artist, description }) => {
  return (
    <aside className={styles.podcast_detail__card}>
      <img
        className={styles.podcast_detail__picture}
        src={image}
        alt={artist}
        aria-label={title}
      />
      <hr className={styles.podcast_detail__separator} />
      <h3 className={styles.podcast_detail__title}>{title}</h3>
      <em className={styles.podcast_detail__artist}>by {artist}</em>
      {description && (
        <>
          <hr className={styles.podcast_detail__separator} />
          <p className={styles.podcast_detail__description}>
            <strong className={styles.podcast_detail__description_strong}>
              Description:{' '}
            </strong>
            <i className={styles.podcast_detail__description_empathized}>
              {description}
            </i>
          </p>
        </>
      )}
    </aside>
  );
};

PodcastDetail.propTypes = {
  image: string,
  title: string,
  artist: string,
  description: string,
};

export default PodcastDetail;
