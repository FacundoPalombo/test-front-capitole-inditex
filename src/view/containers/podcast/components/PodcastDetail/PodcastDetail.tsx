import React from 'react';
import styles from './styles.module.scss';
import { string } from 'prop-types';
import { useParams, Link } from 'react-router-dom';

const PodcastDetail = ({ image, title, artist, description }) => {
  const params = useParams();

  return (
    <section className={styles.podcast_detail}>
      <Link
        aria-label="Go back to podcast detail page."
        to={`/podcast/${params.podcastId}`}
        className={styles.podcast_detail_navigable}
      >
        <aside className={styles.podcast_detail__card}>
          <img
            className={styles.podcast_detail__picture}
            src={image}
            alt={artist}
            aria-label={title}
            loading="lazy"
          />
          <hr className={styles.podcast_detail__separator} />
          <h3 className={styles.podcast_detail__title}>{title}</h3>
          <em className={styles.podcast_detail__artist}>by {artist}</em>
          {description && (
            <>
              <hr className={styles.podcast_detail__separator} />
              <p className={styles.podcast_detail__description}>
                <strong className={styles.podcast_detail__description_strong}>
                  Description:{' '}
                </strong>
                <i className={styles.podcast_detail__description_empathized}>
                  {description}
                </i>
              </p>
            </>
          )}
        </aside>
      </Link>
    </section>
  );
};

PodcastDetail.propTypes = {
  image: string,
  title: string,
  artist: string,
  description: string,
};

export default PodcastDetail;
