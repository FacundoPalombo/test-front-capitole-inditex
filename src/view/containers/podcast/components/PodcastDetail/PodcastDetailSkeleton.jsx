import React from 'react';
import styles from './styles.module.scss';
import image from './asset.jpg';

const PodcastDetailSkeleton = () => {
  return (
    <aside className={styles.podcast_detail__card}>
      <img
        className={styles.podcast_detail__picture}
        src={image}
        alt="..."
        aria-label="Loading Podcast Detail"
      />
      <hr className={styles.podcast_detail__separator} />
      <h3 className={styles.podcast_detail__title}>...</h3>
      <em className={styles.podcast_detail__artist}>...</em>
      <hr className={styles.podcast_detail__separator} />
      <p className={styles.podcast_detail__description}>
        <strong>&nbsp;</strong>
        <i>&nbsp;</i>
      </p>
    </aside>
  );
};

export default PodcastDetailSkeleton;
