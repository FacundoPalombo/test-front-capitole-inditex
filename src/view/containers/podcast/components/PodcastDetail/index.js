export { default } from './PodcastDetail';

import PodcastDetailSkeleton from './PodcastDetailSkeleton';

export { PodcastDetailSkeleton };
