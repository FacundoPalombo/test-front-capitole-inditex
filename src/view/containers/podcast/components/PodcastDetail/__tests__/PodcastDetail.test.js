import React from 'react';
import { render } from '@testing-library/react';
import PodcastDetail from '../PodcastDetail';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: ({ to, children }) => <a href={to}>{children}</a>,
  useParams: jest.fn().mockReturnValue({ podcastId: 1234 }),
}));

describe('PodcastDetail unit tests', () => {
  it('should match snapshot', () => {
    const { asFragment, getByRole } = render(
      <PodcastDetail
        image="imageMock.jpg"
        title="titleMock"
        artist="artistMock"
        description="descriptionMock"
      />
    );
    expect(asFragment()).toMatchSnapshot();
    const navigableComponent = getByRole('link', {
      name: 'titleMock titleMock by artistMock Description: descriptionMock',
    });
    expect(navigableComponent).toHaveAttribute('href', '/podcast/1234');
  });
});
