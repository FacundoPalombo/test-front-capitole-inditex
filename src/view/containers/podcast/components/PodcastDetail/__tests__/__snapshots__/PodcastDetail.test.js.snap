// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastDetail unit tests should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="podcast_detail"
  >
    <a
      href="/podcast/1234"
    >
      <aside
        class="podcast_detail__card"
      >
        <img
          alt="artistMock"
          aria-label="titleMock"
          class="podcast_detail__picture"
          loading="lazy"
          src="imageMock.jpg"
        />
        <hr
          class="podcast_detail__separator"
        />
        <h3
          class="podcast_detail__title"
        >
          titleMock
        </h3>
        <em
          class="podcast_detail__artist"
        >
          by artistMock
        </em>
        <hr
          class="podcast_detail__separator"
        />
        <p
          class="podcast_detail__description"
        >
          <strong
            class="podcast_detail__description_strong"
          >
            Description: 
          </strong>
          <i
            class="podcast_detail__description_empathized"
          >
            descriptionMock
          </i>
        </p>
      </aside>
    </a>
  </section>
</DocumentFragment>
`;
